utils/src/test/kotlin/storage/FileArchiverTest.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.safeMkdirs
+
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.StringSpec
+
+class FileArchiverTest : StringSpec({
+    "All files matching any of the patterns are archived" {
```

Comment:
It seems to me that a lions share of the below test is about testing the `PatchMatcher` class.
Can we instead extract an interface e.g. `FileMatcher` and inject it as dependency?
You could have a `GlobFileMatcher` implementing it and make an instance of it the default parameter value of the constructor param of the `FileArchiver` so that the caller does not necessarily need to mess around with creating an instance.

```
interface `FileMatcher` {
   fun matches(f: File): Boolean
}
```
As result one would not need much more than two files for testing.
No need for (redundantly testing the '*' works in the glob and so on.

Reply:
To me this test is more about testing that the list of patterns is handled correctly by the class, e.g. files are archived if at least one pattern matches, and that the files are copied to the correct locations and have the correct content.

Reply:
@fviernau As discussed offline I will reuse the logic from `LicenseFileMatcher` here, but I would do that in a separate PR if you agree.